projects/igniteui-angular/src/lib/grids/grid/grid.pagination.spec.ts

```diff
@@ -1,557 +1,435 @@
-import { async, TestBed, fakeAsync, tick, ComponentFixture } from '@angular/core/testing';
+import { async, TestBed, fakeAsync, tick } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
 import { IgxGridModule } from './index';
 import {
-    ReorderedColumnsComponent,
-    PagingAndEditingComponent,
-    GridIDNameJobTitleComponent,
-    GridWithUndefinedDataComponent
+    GridWithUndefinedDataComponent, PagingAndEditingComponent
 } from '../../test-utils/grid-samples.spec';
 import { PagingComponent } from '../../test-utils/grid-base-components.spec';
 import { NoopAnimationsModule } from '@angular/platform-browser/animations';
-import { IgxGridComponent } from './grid.component';
 
 import { configureTestSuite } from '../../test-utils/configure-suite';
 import { wait } from '../../test-utils/ui-interactions.spec';
 import { IgxNumberFilteringOperand } from '../../data-operations/filtering-condition';
 import { GridFunctions } from '../../test-utils/grid-functions.spec';
 import { ControlsFunction } from '../../test-utils/controls-functions.spec';
+import { DebugElement } from '@angular/core';
 
 describe('IgxGrid - Grid Paging #grid', () => {
     configureTestSuite();
-    const PAGER_CLASS = '.igx-paginator__pager';
 
     beforeAll(async(() => {
         TestBed.configureTestingModule({
             declarations: [
-                ReorderedColumnsComponent,
                 PagingComponent,
                 PagingAndEditingComponent,
-                GridIDNameJobTitleComponent,
                 GridWithUndefinedDataComponent
             ],
             imports: [IgxGridModule, NoopAnimationsModule]
         }).compileComponents();
     }));
 
-    it('should paginate data UI', fakeAsync(() => {
-        const fix = TestBed.createComponent(PagingComponent);
-        fix.detectChanges();
-        const grid = fix.componentInstance.grid;
-        const gridElement: HTMLElement = fix.nativeElement.querySelector('.igx-grid');
-        const pagingButtons = GridFunctions.getPagingButtons(gridElement);
+    let fix;
+    let grid;
 
-        expect(grid.paging).toBeTruthy();
+    describe('General', () => {
 
-        verifyGridPager(fix, 3, '1', '1\xA0of\xA04', [true, true, false, false]);
+        beforeEach(fakeAsync(() => {
+            fix = TestBed.createComponent(PagingComponent);
+            fix.detectChanges();
+            grid = fix.componentInstance.grid;
+        }));
 
-        // Go to next page
-        pagingButtons[2].dispatchEvent(new Event('click'));
-        tick();
-        fix.detectChanges();
-        verifyGridPager(fix, 3, '4', '2\xA0of\xA04', [false, false, false, false]);
+        afterAll(fakeAsync(() => {
```

Comment:
You can remove this. This is controlled by the configureTestSuite.

Reply:
Done